internal/impl/devbox.go

```diff
@@ -284,6 +287,10 @@ func (d *Devbox) PrintEnv(ctx context.Context, useCache bool, includeHooks bool)
 	envStr := exportify(envs)
 
 	if includeHooks {
+		err = d.writeScriptsToFiles()
+		if err != nil {
+			return "", err
+		}
```

Comment:
I'm not a fan of this change. 

Just like with unified-env, I think we should centralize how we generate the files so that we understand exactly when the files are written. 

This will let us control correctness, and can also optimize that codepath via caching-and-invalidation.

Reply:
please see my suggestion above for how we could ensure the `writeScriptsToFiles` is called from within `ensurePackagesAreInstalled`.